src/fight/mod.rs

```diff
@@ -3,156 +3,301 @@
+fn compute_case(
+    attacking_regiment: &regiment::Regiment,
+    defending_regiment: &regiment::Regiment,
+    case: ComputeCase,
+) -> (usize, f64) {
+    let attacking_model: &model::Model = attacking_regiment.get_model();
+    let defending_model: &model::Model = defending_regiment.get_model();
+    let nb_touch: usize = (attacking_model.get_stats().get_attack() as f64
+        * 1.5
+        * attacking_regiment.get_cols() as f64)
+        .round() as usize;
+    let touch_probability: f64 =
+        computation_tools::compute_damage_probability(attacking_model, defending_model);
+    let mut results: Vec<(usize, f64)> = Vec::new();
+    let mut selection: Vec<(usize, f64)> = Vec::new();
+    let mut bondaries: (isize, isize);
+    let threshold: f64 = match case {
+        ComputeCase::BEST => global_values::BEST_CASE_THRESHOLD,
+        ComputeCase::WORST => global_values::WORST_CASE_THRESHOLD,
+        _ => 0.0_f64,
+    };
+    let defender_hp = defending_model.get_stats().get_health_point();
+
+    if let ComputeCase::MEAN = case {
+        return compute_mean_case(attacking_regiment, defending_regiment);
+    }
+    for i in 0..nb_touch {
+        bondaries = match case {
+            ComputeCase::BEST => (i as isize, nb_touch as isize),
+            ComputeCase::WORST => (0, i as isize),
+            ComputeCase::MEAN => unreachable!("Code not supposed to be reached!"),
+        };
+        results.push((
+            i,
+            math_tools::sigma(
+                bondaries.0,
+                bondaries.1,
+                |curr, param, _, _| {
+                    math_tools::compute_bernoulli(
+                        param.unwrap().1 as usize,
+                        curr as usize,
+                        param.unwrap().0,
+                    )
+                },
+                Some((touch_probability, nb_touch)),
+            ),
+        ));
+    }
+
+    for i in results.iter() {
+        if i.1 >= threshold {
+            selection.push(*i);
+        }
+    }
+
+    if selection.is_empty() {
+        let size = (0.1 * results.len() as f64).floor() as usize;
+        for i in results {
+            if selection.len() < size {
+                selection.push(i);
+            } else {
+                match selection.iter().position(|e| e.1 < i.1) {
+                    Some(x) => selection[x] = i,
+                    None => (),
+                }
+            }
+        }
+    }
+
+    let res: (usize, f64) = match case {
+        ComputeCase::BEST => selection.into_iter().max_by_key(|e| e.0).unwrap(),
+        ComputeCase::WORST => selection.into_iter().min_by_key(|e| e.0).unwrap(),
+        ComputeCase::MEAN => unreachable!("Code not supposed to be reached!"),
+    };
+    bondaries = match case {
+        ComputeCase::BEST => (defender_hp as isize, nb_touch as isize),
+        ComputeCase::WORST => (0, defender_hp as isize),
+        ComputeCase::MEAN => unreachable!("Code not supposed to be reached!"),
```

Comment:
I have a function that handle the ComputeCase::MEAN because it is not computed on the same way than the two others. At the top of the function I return compute_mean_case() if case == ComputeCase::MEAN
So basically I'll throw some holy water on my computer if case is ComputeCase::MEAN at this point